src/sentry/lang/native/symbolicator.py

```diff
@@ -192,6 +192,96 @@ def process_payload(self, stacktraces, modules, signal=None):
         )
 
 
+def redact_internal_sources(response):
+    """Redacts information about internal sources from a response.
+
+    Symbolicator responses can contain a section about DIF object file candidates where were
+    attempted to be downloaded from the sources.  This includes a full URI of where the
+    download was attempted from.  For internal sources we want to redact this in order to
+    not leak any internal details.
+
+    Note that this modifies the argument passed in, thus redacting in-place.  It still
+    returns the modified response.
+    """
+    for module in response.get("modules", []):
+        redact_internal_sources_from_module(module)
+    return response
+
+
+def sentry_dsyms_id(url):
+    """Extracts the ID of the debug symbol for this project from the location URI.
+
+    The get_internal_source() function returns URLs to the local sentry server, symbolicator
+    then adds a query string and the location returned in a object dif candidate is
+    something like
+    ``http::servername/api/<project_id>/projects/<org_slug>/<project_slug>/files/dsyms/?id=<id>``.
+    This parses out the ``<id>`` from this.
+    """
+    parsed = urlparse(url)
+    qs = parse_qs(parsed.query)
+    return qs["id"][0]
```

Comment:
From an offline conversation, and please take with a grain of salt as I haven't done a full review yet: 
Since we have a dedicated `sentry` source type, we could generate the logical URIs as location in Symbolicator. That would reduce the coupling between the URLs that symbolicator happens to use for downloading and the code here that sanitizes them mainly for the UI.

Reply:
Yep, gone with that as it simplifies things here and does seem reasonable from symbolicator's point of view as well:https://github.com/getsentry/symbolicator/pull/358